### PR TITLE
add docs for `go_deps.gazelle_default_attributes`

### PR DESCRIPTION
**Pull request by tyler-french**
This PR adds missing documentation

### CODE REVIEW

File: docs/go/core/bzlmod.md [hunk 1]
```diff
@@ -314,5 +338,5 @@ go_deps.archive_override(
 
 ### Not yet supported
 
-* Fetching dependencies from Git repositories
-* `go.mod` `exclude` directices
+-   Fetching dependencies from Git repositories
```
Comment (tyler-french) [inline docs/go/core/bzlmod.md:341]: not sure why these changes are being added to this PR 🤔

Comment (tyler-french) [reply docs/go/core/bzlmod.md:341]: Looks like it's coming from a linter 🤷🏻

File: docs/go/core/bzlmod.md [hunk 2]
```diff
@@ -183,8 +183,9 @@ use_repo(
 ```
 
 Limitations:
-* `go.work` is supported exclusively in the root module.
-* Dependencies that are indirect and depend on a go module specified in `go.work` will have that dependency diverge from the one in `go.work`. More details can be found here: https://github.com/bazelbuild/bazel-gazelle/issues/1797.
+
```
Comment (tyler-french) [inline docs/go/core/bzlmod.md:186]: These changes are also coming from some IDE linter

File: docs/go/core/bzlmod.md [hunk 3]
```diff
@@ -244,7 +245,7 @@ go_deps.config(
 ```
 
 Variables set in this way are used by `go_deps` as well as `@rules_go//go`, with other variables inheriting their value from the host environment.
-`go_env` does *not* affect Go build actions.
+`go_env` does _not_ affect Go build actions.
```
Comment (fmeum) [inline docs/go/core/bzlmod.md:248]:
```suggestion
`go_env` does **no** affect Go build actions.
```

draws more attention now that we are changing this anyway

File: docs/go/core/bzlmod.md [hunk 4]
```diff
@@ -279,6 +280,29 @@ go_deps.gazelle_override(
 If you need to use a `gazelle_override` to get a public Go module to build with Bazel, consider contributing the directives to the [public registry for default Gazelle overrides](https://github.com/bazelbuild/bazel-gazelle/blob/master/internal/bzlmod/default_gazelle_overrides.bzl) via a PR.
 This will allow you to drop the `gazelle_override` tag and also makes the Go module usable in non-root Bazel modules.
 
+Users can apply custom default directives or extra args to **all** modules, these can be added via a `go_deps.gazelle_default_attributes`. These will
+disable/overwrite the
```
Comment (fmeum) [inline docs/go/core/bzlmod.md:284]: This sentence is incomplete.

File: docs/go/core/bzlmod.md [hunk 5]
```diff
@@ -279,6 +280,29 @@ go_deps.gazelle_override(
...
+        "-go_naming_convention_external=go_default_library",
+    ],
+    build_file_generation = "on",
+    directives = [
+        "gazelle:proto disable",
+    ],
+)
+```
+
+Overrides are applied in this order precedence:
```
Comment (fmeum) [inline docs/go/core/bzlmod.md:298]:
```suggestion
Overrides are applied with precedence decreasing in this order:
```
(or similar, but the original is certainly missing words)

File: docs/go/core/bzlmod.md [hunk 6]
```diff
@@ -244,7 +245,7 @@ go_deps.config(
 ```
 
 Variables set in this way are used by `go_deps` as well as `@rules_go//go`, with other variables inheriting their value from the host environment.
-`go_env` does *not* affect Go build actions.
+`go_env` does **no** affect Go build actions.
```
Comment (fmeum) [inline docs/go/core/bzlmod.md:248]:
```suggestion
`go_env` does **not** affect Go build actions.
```

**Review by fmeum — APPROVED** [review]
